### PR TITLE
Made AbstractKubernetesDeployer methods protected for extension.

### PR DESCRIPTION
**Pull request by donovanmuller**
Bumped kubernetes-client version to latest. Extracted getExternalPort for reuse.

Address #39

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java [hunk 1]
```diff
@@ -186,6 +181,16 @@ public AppStatus status(String appId) {
 		return status;
 	}
 
+	protected int getExternalPort(final AppDeploymentRequest request) {
```
Comment (trisberg) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: this isn't really a "getter" method, maybe name it configureExternalPort

**Comment by trisberg** [comment]
Thanks for the PR, making this more extensible makes sense. Have you signed our contributor agreement - https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-docs/src/main/asciidoc/appendix-contributing.adoc

**Comment by donovanmuller** [comment]
Yes, I've [signed before](https://github.com/spring-cloud/spring-cloud-dataflow/pull/723#issuecomment-227709450). I'll make the [change](https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/pull/40#discussion_r70217009) quick and update...

**Comment by donovanmuller** [comment]
@trisberg if all good, any chance this can be merged 😄

**Comment by trisberg** [comment]
Having trouble using the 1.3.98 version of kubernetes-client, get some errors while running the tests. Although the build succeeds, there is definitely something that has changed and our code doesn't work as expected.

```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.fabric8.kubernetes.client.Adapters
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:57) ~[kubernetes-client-1.3.98.jar:na]
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:41) ~[kubernetes-client-1.3.98.jar:na]
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:37) ~[kubernetes-client-1.3.98.jar:na]
	at io.fabric8.kubernetes.client.DefaultKubernetesClient.<init>(DefaultKubernetesClient.java:99) ~[kubernetes-client-1.3.98.jar:na]
	at org.springframework.cloud.deployer.spi.kubernetes.KubernetesAutoConfiguration.kubernetesClient(KubernetesAutoConfiguration.java:59) ~[classes/:na]
	at org.springframework.cloud.deployer.spi.kubernetes.KubernetesAutoConfiguration$$EnhancerBySpringCGLIB$$f3011f1f.CGLIB$kubernetesClient$2(<generated>) ~[classes/:na]
	at org.springframework.cloud.deployer.spi.kubernetes.KubernetesAutoConfiguration$$EnhancerBySpringCGLIB$$f3011f1f$$FastClassBySpringCGLIB$$41c9d87c.invoke(<generated>) ~[classes/:na]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228) ~[spring-core-4.2.6.RELEASE.jar:4.2.6.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356) ~[spring-context-4.2.6.RELEASE.jar:4.2.6.RELEASE]
	at org.springframework.cloud.deployer.spi.kubernetes.KubernetesAutoConfiguration$$EnhancerBySpringCGLIB$$f3011f1f.kubernetesClient(<generated>) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_40]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_40]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_40]
	at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_40]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.2.6.RELEASE.jar:4.2.6.RELEASE]
	... 38 common frames omitted
```

**Comment by trisberg** [comment]
Rebased and squashed, merged as commit 634ce96dc10383b88c17ceaa2e75582005122c5c

Removed the version update - let's do that in a separate commit

**Pull request closed without merging**